fastanvil/src/complete/chunk.rs

```diff
@@ -0,0 +1,56 @@
+use std::ops::Range;
+
+use crate::biome::Biome;
+use crate::complete::section_tower::SectionTower;
+use crate::{dimension, Block, CurrentJavaChunk, HeightMode, JavaChunk};
+
+pub struct Chunk {
+    pub status: String,
+
+    pub sections: SectionTower,
+}
+
+impl Chunk {
+    pub fn from_bytes(data: &[u8]) -> fastnbt::error::Result<Self> {
+        return match JavaChunk::from_bytes(data)? {
+            JavaChunk::Post18(chunk) => Ok(Self::from_current_chunk(&chunk)),
+            JavaChunk::Pre18(_) => {
+                todo!()
+            }
+            JavaChunk::Pre13(_) => {
+                todo!()
+            }
+        };
+    }
+
+    pub fn from_current_chunk(current_java_chunk: &CurrentJavaChunk) -> Self {
```

Comment:
This can take `CurrentJavaChunk` by value rather than reference. The caller can clone it if necessary.

Reply:
may i ask where the advantage would be?

Reply:
Of course. The caller may not care about the `CurrentJavaChunk` once they have converted to a `complete::Chunk`. Taking it by reference forces copying various data when it could simply be passed by value and moved.

Taking it by value moves the choice to the caller. If they want to maintain a `CurrentJavaChunk` they clone before passing it by value. If we take by reference the caller has no choice, it's (or bits of it are) always cloned.

Reply:
Of course. The caller may not care about the `CurrentJavaChunk` once they have converted to a `complete::Chunk`. Taking it by reference forces copying various data when it could simply be passed by value and moved.

Taking it by value moves the choice to the caller. If they want to maintain a `CurrentJavaChunk` they clone before passing it by value. If we take by reference the caller has no choice, it's (or bits of it are) always cloned.